### ISSUE TITLE
Announcement: Post-IETF 114 and the future of this Draft

### ISSUE THREAD

**Issue by kyzer-davis** [uuid6/uuid6-ietf-draft#122]
Hello All,

@bradleypeabody and I recently represented [Draft 04 of the UUIDv6/7/8/Max Draft](https://datatracker.ietf.org/doc/draft-peabody-dispatch-new-uuid-format/04/) in the last IETF 114 Dispatch session back on July 25th, 2022

The goal of the Dispatch session was to present the work we have already done up through Draft 04 and obtain an official dispatch outcome.
That is, 
- It goes to an existing working group
- It's simple enough to be AD-sponsored (i.e. doesn't need lots of community discussion and regular meetings) 
- It requires a new working group 
- It goes to ISE (Independent Submissions Editor) because you don't want community input
- It needs more work before finding a home

---
### The Outcome
- The IETF dispatch opted to officially adopt our work through a new "tightly focused" UUID working group (WG) which is committed to ensuring UUID is one of the IETFs "well maintained standards".
- The work in progress charter can be found at this link: https://datatracker.ietf.org/doc/charter-ietf-uuidrev/
  - Note: 00-03 is the charter version at the time of this announcement: https://www.ietf.org/charter/charter-ietf-uuidrev-00-03.txt

---
### The WG Charter
- The most important item to take away from the charter:
   - The IETF wants to give RFC4122 a "fresh coat of paint" and replace it with with a net-new document.
- In the process we can incorporate the v6/7/8/Max into the new base UUID RFC.
- This means things like #119 and other [RFC4122 errata's](https://www.rfc-editor.org/errata/rfc4122) can finally be fixed!
- Other Notes: The IETF is not opposed to handling other topics in this WG like [Alternate UUID Encoding or UUID Long](https://github.com/uuid6/new-uuid-encoding-techniques-ietf-draft); however these topics are contingent on delivering the RFC4122 re-write.

---
### What's Next?
- The charter for the WG creation will be reviewed by the [Internet Engineering Steering Group (IESG) on Aug 25](https://datatracker.ietf.org/iesg/agenda/documents/#autoid-15) for official adoption by the IETF.
- Assuming this WG is adopted @bradleypeabody, Myself and possibly @ben221199 will begin working on the draft which will define/redefine UUIDv1 through UUIDv8, Nil/Max and replace RFC4122.
- I will continue to update this thread as I have information to share.

---
### IETF 114 Links
- [IETF 114 Dispatch Meeting Minutes](https://datatracker.ietf.org/doc/minutes-114-dispatch-202207251000/#new-uuid-formats-20-min)
- [IETF 114 Dispatch Recording](https://youtu.be/YyykUxLQhrc?t=3501) [Start:  0:58:21] - [Stop: 1:11:34] _(Ignore me failing to use their mic setup)_
- [Session Slides](https://docs.google.com/presentation/d/1zrKS5x8yN0by_wkyq0nz6nm4UjwoHAPHGRsmg1s2jKE/edit?usp=sharing)

---

Thanks again to all who have contributed thus far!

**Comment by kyzer-davis** [comment]
There is a new mailer to track this work over on the IETF if members here want to subscribe so they can contribute any future discussions or watch silently.
https://www.ietf.org/mailman/listinfo/uuidrev

I will also work to ensure any major topics are properly tracked on the issue tracker in this forum.

**Comment by kyzer-davis** [comment]
The Working Group has been accepted and work shall commence on `draft-ietf-uuidrev-rfc4122bis-00`
https://datatracker.ietf.org/wg/uuidrev/about/

Please subscribe to the mailer to stay in the loop in case I forget to publish something to this announcement thread.

Edit:
Historical email communications can be found here: https://mailarchive.ietf.org/arch/browse/uuidrev/

**Comment by kyzer-davis** [comment]
Timeline Update from last IETF Interim Meeting:
- Submit Draft 00 of draft-ietf-uuidrev-rfc4122bis by Oct 24 (Draft 04 merged with RFC4122 is the goal)
- Discuss at IETF 115, Week of Nov 5
- Interim Meeting every month, draft release at the end of every month
- Goal still to submit to IESG end of March after IETF 116

**Comment by kyzer-davis** [comment]
```
Name:		draft-ietf-uuidrev-rfc4122bis
Revision:	00
Title:		A Universally Unique IDentifier (UUID) URN Namespace
Document date:	2022-10-17
Group:		uuidrev
Pages:		61
URL:            https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-00.txt
Status:         https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
Html:           https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-00.html
Htmlized:       https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
```

Edit:
```
Diffs:          https://www.ietf.org/rfcdiff?url2=draft-ietf-uuidrev-rfc4122bis-00
```

**Comment by kyzer-davis** [comment]
At IETF 115 it was decided to have some interim meetings leading up to Draft 01 to discuss changes.

Meeting info for those that want to attend are on Jan 19, 2023 and Feb 16, 2023:
https://datatracker.ietf.org/wg/uuidrev/meetings/

You may also find historical updates from IETF 115 and the interim meeting in 2022 on the same page.

**Comment by kyzer-davis** [comment]
```
Name:		draft-ietf-uuidrev-rfc4122bis
Revision:	01
Title:		A Universally Unique IDentifier (UUID) URN Namespace
Document date:	2023-01-23
Group:		uuidrev
Pages:		62
URL:            https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-01.txt
Status:         https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
Html:           https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-01.html
Htmlized:       https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
Diff:           https://author-tools.ietf.org/iddiff?url2=draft-ietf-uuidrev-rfc4122bis-01
```

**Comment by kyzer-davis** [comment]
```
Name:		draft-ietf-uuidrev-rfc4122bis
Revision:	02
Title:		A Universally Unique IDentifier (UUID) URN Namespace
Document date:	2023-02-16
Group:		uuidrev
Pages:		47
URL:            https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-02.txt
Status:         https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
Html:           https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-02.html
Htmlized:       https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
Diff:           https://author-tools.ietf.org/iddiff?url2=draft-ietf-uuidrev-rfc4122bis-02
```

**Comment by sergeyprokhorenko** [comment]
Hello All,
Typo: missing d in words "(as opposed to client-generate**d** UUIDs)"

**Comment by kyzer-davis** [comment]
```
Name:		draft-ietf-uuidrev-rfc4122bis
Revision:	03
Title:		Universally Unique IDentifiers (UUID)
Document date:	2023-04-12
Group:		uuidrev
Pages:		48
URL:            https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-03.txt
Status:         https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
Html:           https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-03.html
Htmlized:       https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
Diff:           https://author-tools.ietf.org/iddiff?url2=draft-ietf-uuidrev-rfc4122bis-03
```

**Comment by kyzer-davis** [comment]
### interim-2023-uuidrev-03 2023-05-10 Update:
`draft-ietf-uuidrev-rfc4122bis-03` has entered Working Group Last Call!
Please comment on the tracker or email thread before June 9th to request any changes.

The Mailer: [uuidrev@ietf.org](mailto:uuidrev@ietf.org) ([signup required to post](https://www.ietf.org/mailman/listinfo/uuidrev))
IETF Github: https://github.com/ietf-wg-uuidrev/rfc4122bis/issues

**Comment by kyzer-davis** [comment]
### WGCL Draft 04 Midterm Update

```
Name:		draft-ietf-uuidrev-rfc4122bis
Revision:	04
Title:		Universally Unique IDentifiers (UUID)
Document date:	2023-05-23
Group:		uuidrev
Pages:		49
URL:            https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-04.txt
Status:         https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
Html:           https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-04.html
Htmlized:       https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
Diff:           https://author-tools.ietf.org/iddiff?url2=draft-ietf-uuidrev-rfc4122bis-04
```

**Comment by kyzer-davis** [comment]
```
Name:		draft-ietf-uuidrev-rfc4122bis
Revision:	05
Title:		Universally Unique IDentifiers (UUID)
Document date:	2023-06-02
Group:		uuidrev
Pages:		49
URL:            https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-05.txt
Status:         https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
Html:           https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-05.html
Htmlized:       https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
Diff:           https://author-tools.ietf.org/iddiff?url2=draft-ietf-uuidrev-rfc4122bis-05
```

**Comment by kyzer-davis** [comment]
```
Name:		draft-ietf-uuidrev-rfc4122bis
Revision:	06
Title:		Universally Unique IDentifiers (UUID)
Document date:	2023-06-06
Group:		uuidrev
Pages:		49
URL:            https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-06.txt
Status:         https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
Html:           https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-06.html
Htmlized:       https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
Diff:           https://author-tools.ietf.org/iddiff?url2=draft-ietf-uuidrev-rfc4122bis-06
```

**Comment by kyzer-davis** [comment]
```
Name:		draft-ietf-uuidrev-rfc4122bis
Revision:	07
Title:		Universally Unique IDentifiers (UUID)
Document date:	2023-06-09
Group:		uuidrev
Pages:		49
URL:            https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-07.txt
Status:         https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
Html:           https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-07.html
Htmlized:       https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
Diff:           https://author-tools.ietf.org/iddiff?url2=draft-ietf-uuidrev-rfc4122bis-07
```

**Comment by kyzer-davis** [comment]
```
Name:		draft-ietf-uuidrev-rfc4122bis
Revision:	08
Title:		Universally Unique IDentifiers (UUID)
Document date:	2023-07-24
Group:		uuidrev
Pages:		50
URL:            https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-08.txt
Status:         https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
Html:           https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-08.html
Htmlized:       https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
Diff:           https://author-tools.ietf.org/iddiff?url2=draft-ietf-uuidrev-rfc4122bis-08
```

**Comment by kyzer-davis** [comment]
```
Name:		draft-ietf-uuidrev-rfc4122bis
Revision:	09
Title:		Universally Unique IDentifiers (UUID)
Document date:	2023-08-04
Group:		uuidrev
Pages:		51
URL:            https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-09.txt
Status:         https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
Html:           https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-09.html
Htmlized:       https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
Diff:           https://author-tools.ietf.org/iddiff?url2=draft-ietf-uuidrev-rfc4122bis-09
```

**Comment by kyzer-davis** [comment]
```
Name:		draft-ietf-uuidrev-rfc4122bis
Revision:	10
Title:		Universally Unique IDentifiers (UUID)
Document date:	2023-08-23
Group:		uuidrev
Pages:		51
URL:            https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-10.txt
Status:         https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
Html:           https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-10.html
Htmlized:       https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
Diff:           https://author-tools.ietf.org/iddiff?url2=draft-ietf-uuidrev-rfc4122bis-10
```

**Comment by kyzer-davis** [comment]
```
Name:     draft-ietf-uuidrev-rfc4122bis
Revision: 11
Title:    Universally Unique IDentifiers (UUID)
Date:     2023-09-05
Group:    uuidrev
Pages:    52
URL:      https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-11.txt
Status:   https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
HTML:     https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-11.html
HTMLized: https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
Diff:     https://author-tools.ietf.org/iddiff?url2=draft-ietf-uuidrev-rfc4122bis-11
```

**Comment by kyzer-davis** [comment]
```
Name:     draft-ietf-uuidrev-rfc4122bis
Revision: 12
Title:    Universally Unique IDentifiers (UUID)
Date:     2023-10-05
Group:    uuidrev
Pages:    55
URL:      https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-12.txt
Status:   https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
HTML:     https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-12.html
HTMLized: https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
Diff:     https://author-tools.ietf.org/iddiff?url2=draft-ietf-uuidrev-rfc4122bis-12
```

**Comment by kyzer-davis** [comment]
```
Name:     draft-ietf-uuidrev-rfc4122bis
Revision: 13
Title:    Universally Unique IDentifiers (UUID)
Date:     2023-10-20
Group:    uuidrev
Pages:    58
URL:      https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-13.txt
Status:   https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
HTML:     https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-13.html
HTMLized: https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
Diff:     https://author-tools.ietf.org/iddiff?url2=draft-ietf-uuidrev-rfc4122bis-13
```

**Comment by kyzer-davis** [comment]
```
Name:     draft-ietf-uuidrev-rfc4122bis
Revision: 14
Title:    Universally Unique IDentifiers (UUID)
Date:     2023-11-06
Group:    uuidrev
Pages:    58
URL:      https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-14.txt
Status:   https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/
HTML:     https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-14.html
HTMLized: https://datatracker.ietf.org/doc/html/draft-ietf-uuidrev-rfc4122bis
Diff:     https://author-tools.ietf.org/iddiff?url2=draft-ietf-uuidrev-rfc4122bis-14
```

**Comment by kyzer-davis** [comment]
```
RFC 9562

Title:      Universally Unique IDentifiers (UUIDs) 
Author:     K. Davis,
            B. Peabody,
            P. Leach
Status:     Standards Track
Stream:     IETF
Date:       May 2024
Mailbox:    kydavis@cisco.com,
            brad@peabody.io,
            pjl7@uw.edu
Pages:      46
Obsoletes:  RFC 4122

I-D Tag:    draft-ietf-uuidrev-rfc4122bis-14.txt

URL:        https://www.rfc-editor.org/info/rfc9562

DOI:        10.17487/RFC9562
```

With the final publication I plan to archive this Repo.
The IETF working group list ([uuidrev@ietf.org](mailto:uuidrev@ietf.org)) shall remain open to address any questions, or errata that may arise in the future.

**Comment by broofa** [comment]
Congratulations, and thanks for all the hard work!  (And persistence :-) )

**Comment by htunnicliff** [comment]
Thank you so much!! 👏

**Comment by ramsey** [comment]
Congratulations! Thank you for all the hard work to see this through. It looks great!

**Comment by fabiolimace** [comment]
Congratulations @bradleypeabody and @kyzer-davis! Thank you for your dedication to the specification, which resulted in a high quality document, technically and textually. The Internet as a whole will benefit greatly from the new published standard. As for the community that actively participated during the project by discussing, giving suggestions and making prototypes, it was great exchanging ideas with you guys! I will miss the discussions.
